cedar-policy-validator/src/schema_file_format.rs

```diff
@@ -52,6 +52,16 @@ pub struct SchemaFragment(
     pub  HashMap<Option<Name>, NamespaceDefinition>,
 );
 
+impl SchemaFragment {
+    /// Resolve common types in the schema
+    pub(crate) fn resolve_common_types(self) -> Result<Self> {
+        let mut schema = self.clone();
```

Comment:
Is `.clone()` necessary?  Cloning the whole schema seems like something that could eventually be a performance problem